pkg/operator/controller/nodeobservability/controller.go

```diff
@@ -335,8 +339,41 @@ func (r *NodeObservabilityReconciler) ensureNodeObservabilityDeleted(ctx context
 
 // machineConfigChangeRequested returns true, when a given NodeObservabilityType needs
 // machine config change. Only CrioKubeletNodeObservabilityType requires a MC change, false otherwise
-func machineConfigChangeRequested(nodeObs *operatorv1alpha2.NodeObservability) bool {
-	return nodeObs.Spec.Type == operatorv1alpha2.CrioKubeletNodeObservabilityType
+func (r *NodeObservabilityReconciler) machineConfigChangeRequested(ctx context.Context, nodeObs *operatorv1alpha2.NodeObservability) bool {
+	isChangeRequested := nodeObs.Spec.Type == operatorv1alpha2.CrioKubeletNodeObservabilityType
+	if !isChangeRequested {
+		if nodeObs.Spec.Type != "" {
+			r.Log.Error(nil, "unsupoorted node observability type found", "type", nodeObs.Spec.Type)
+		}
```

Comment:
This is unnecessary as there cannot be any other type for the moment: [openapi validation](https://github.com/openshift/node-observability-operator/blob/319c83f0300b327641a3f39d405a952b67f49e03/api/v1alpha2/nodeobservability_types.go#L23-L24).

Reply:
Addressed